robot-server/robot_server/instruments/instrument_models.py

```diff
@@ -0,0 +1,105 @@
+"""Request and response models for /instruments endpoints."""
+from __future__ import annotations
+
+from typing_extensions import Literal
+from typing import Optional, TypeVar, Union, Generic
+from enum import Enum
+from pydantic import BaseModel, Field
+from pydantic.generics import GenericModel
+
+from opentrons.hardware_control.instruments.ot3.instrument_calibration import (
+    GripperCalibrationOffset,
+)
+from opentrons.hardware_control.types import GripperJawState
+from opentrons.types import Mount
+
+from opentrons_shared_data.pipette.dev_types import (
+    PipetteName,
+    PipetteModel,
+    ChannelCount,
+)
+from opentrons_shared_data.gripper.dev_types import (
+    GripperName,
+    GripperModel,
+)
+
+InstrumentNameT = TypeVar("InstrumentNameT", bound=Union[GripperName, PipetteName])
+InstrumentModelT = TypeVar("InstrumentModelT", bound=Union[GripperModel, PipetteModel])
+InstrumentDataT = TypeVar("InstrumentDataT", bound=BaseModel)
+
+InstrumentType = Literal["pipette", "gripper"]
+
+
+# TODO (spp, 2023-01-03): use MountType from opentrons.types once it has extension type
+class MountType(str, Enum):
+    """Available mount types."""
+
+    LEFT = "left"
+    RIGHT = "right"
+    EXTENSION = "extension"
+
+    @staticmethod
+    def from_hw_mount(mount: Mount) -> MountType:
+        """Convert from Mount to MountType."""
+        mount_map = {Mount.LEFT: MountType.LEFT, Mount.RIGHT: MountType.RIGHT}
+        return mount_map[mount]
```

Comment:
I suggest making this an `if`-`elif` ladder would let mypy check that we've exhaustively enumerated every possibility of `Mount`. So when `Mount` gains an "extension" value, this will become a type-checking error and prompt us to update this code.

```suggestion
        if mount == mount.LEFT:
            return MountType.LEFT
        elif mount == mount.RIGHT:
            return MountType.RIGHT
```